docker/vendor/fluentd/Dockerfile

```diff
@@ -13,11 +13,57 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM gcr.io/google-containers/fluentd-elasticsearch:v2.2.0
+FROM debian:sid-slim
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+COPY include/Gemfile /Gemfile
+
+# 1. Install & configure dependencies.
+# 2. Install fluentd via ruby.
+# 3. Remove build dependencies.
+# 4. Cleanup leftover caches & files.
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+                     make \
+                     gcc \
+                     g++ \
+                     libc6-dev \
+                     ruby-dev \
+                     libffi-dev \
+                     ca-certificates \
+                     libjemalloc2 \
+                     ruby \
+    && echo 'gem: --no-document' >> /etc/gemrc \
+    && gem install --file Gemfile \
+    && apt-get purge -y --auto-remove \
+                     -o APT::AutoRemove::RecommendsImportant=false \
+                     $BUILD_DEPS \
```

Comment:
The base layer had:

```
RUN BUILD_DEPS="make gcc g++ libc6-dev ruby-dev libffi-dev" \
```

but we don't set that, so those build deps are never removed.